sql: TestScrubFKConstraintFKMissing failed

sql.TestScrubFKConstraintFKMissing [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7919720?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7919720?buildTab=artifacts#/) on release-22.2 @ [ff69f981a02ea6906d4be138189c4531a6a420d1](https://github.com/cockroachdb/cockroach/commits/ff69f981a02ea6906d4be138189c4531a6a420d1):


```
=== RUN   TestScrubFKConstraintFKMissing
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestScrubFKConstraintFKMissing502055319
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    scrub_test.go:605: unexpected error: pq: scrub-unique: inconsistent AS OF SYSTEM TIME timestamp; expected: 1670758122.599437696,1, got: 1670758122.599437696,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestScrubFKConstraintFKMissing502055319
--- FAIL: TestScrubFKConstraintFKMissing (1.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScrubFKConstraintFKMissing.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22313

I was able to reproduce this with stress on a gceworker on tip of master after about 7 hours:

```
michae2@gceworker-michae2:~/go/src/github.com/cockroachdb/cockroach$ ./dev test pkg/sql -f TestScrubFKConstraintFKMissing --stress
$ bazel test //pkg/sql:sql_test --test_env=GOTRACEBACK=all --test_timeout=86400 --test_env=COCKROACH_STRESS=true --run_under '@com_github_cockroachdb_stress//:stress -bazel -shardable-artifacts '"'"'XML_OUTPUT_FILE=/home/michae2/go/src/github.com/cockroachdb/cockroach/bin/dev-versions/dev.67 merge-test-xmls'"'"' ' --test_filter=TestScrubFKConstraintFKMissing --test_sharding_strategy=disabled --test_output streamed
INFO: Invocation ID: f1cd79b0-c5ac-479c-98ae-feb25c4de983
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Build options --//build/toolchains:crdb_test_flag, --run_under, and --test_env have changed, discarding analysis cache.
INFO: Analyzed target //pkg/sql:sql_test (25 packages loaded, 24100 targets configured).
INFO: Found 1 test target...
66 runs so far, 0 failures, over 5s
137 runs so far, 0 failures, over 10s
...

384754 runs so far, 0 failures, over 7h5m30s
384828 runs so far, 0 failures, over 7h5m35s

initialized metamorphic constant "span-reuse-rate" with value 24
initialized metamorphic constant "parse-json-impl" with value 1
initialized metamorphic constant "mvcc-value-disable-simple-encoding" with value true
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 1
initialized metamorphic constant "coldata-batch-size" with value 2106
initialized metamorphic constant "default-batch-bytes-limit" with value 24792
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "inverted-joiner-batch-size" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "spilling-queue-initial-len" with value 10
initialized metamorphic constant "merge-joiner-groups-buffer" with value 10
initialized metamorphic constant "ColIndexJoin-batch-size" with value 3103559
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 1504018
initialized metamorphic constant "max-batch-size" with value 1145
initialized metamorphic constant "max-batch-byte-size" with value 14402123
initialized metamorphic constant "parallel-scan-result-threshold" with value 6432
initialized metamorphic constant "addsst-rewrite-concurrency" with value 12
initialized metamorphic constant "split-scans-right-for-stats-first" with value true
initialized metamorphic constant "copy-batch-size" with value 8461
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 10
initialized metamorphic constant "use-index-lookup-for-descriptors-in-database" with value false
initialized metamorphic constant "copy-fast-path-enabled-default" with value false
initialized metamorphic constant "span-reuse-rate" with value 57
initialized metamorphic constant "parse-json-impl" with value 0
initialized metamorphic constant "kv.rangefeed.use_dedicated_connection_class.enabled" with value true
initialized metamorphic constant "mvcc-incremental-iter-tbi" with value false
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 0
initialized metamorphic constant "default-batch-bytes-limit" with value 64872
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "datum-row-converter-batch-size" with value 1
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "spilling-queue-initial-len" with value 10
initialized metamorphic constant "merge-joiner-groups-buffer" with value 14
initialized metamorphic constant "ColIndexJoin-batch-size" with value 185738
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 664203
initialized metamorphic constant "max-batch-size" with value 8128
initialized metamorphic constant "max-batch-byte-size" with value 17941832
initialized metamorphic constant "parallel-scan-result-threshold" with value 8346
initialized metamorphic constant "addsst-rewrite-concurrency" with value 4
initialized metamorphic constant "split-scans-right-for-stats-first" with value true
initialized metamorphic constant "resolve-membership-single-scan-enabled" with value false
initialized metamorphic constant "copy-batch-size" with value 2941
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 23
initialized metamorphic constant "copy-fast-path-enabled-default" with value false
I221217 08:20:01.713244 1 (gostd) rand.go:199  [-] 1  random seed: 7411311068975742005
--- FAIL: TestScrubFKConstraintFKMissing (0.70s)
    test_log_scope.go:161: test logs captured to: /home/michae2/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestScrubFKConstraintFKMissing1974980698
    test_log_scope.go:79: use -show-logs to present logs inline
    scrub_test.go:605: unexpected error: pq: scrub-unique: inconsistent AS OF SYSTEM TIME timestamp; expected: 1671265202.394072788,1, got: 1671265202.394072788,0
    panic.go:522: -- test log scope end --
FAIL
test logs left over in: /home/michae2/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestScrubFKConstraintFKMissing1974980698
I221217 08:20:02.428708 1 (gostd) testmain.go:960  [-] 1  Test //pkg/sql:sql_test exited with error code 1


ERROR: exit status 1

384846 runs completed, 1 failures, over 7h5m36s
context canceled
FAIL
FAIL: //pkg/sql:sql_test (see /home/michae2/.cache/bazel/_bazel_michae2/88dadb79cb9eca13f5ad7e068969d8c1/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/sql/sql_test/test.log)
Target //pkg/sql:sql_test up-to-date:
  _bazel/bin/pkg/sql/sql_test_/sql_test
INFO: Elapsed time: 25632.511s, Critical Path: 25630.07s
INFO: 457 processes: 3 internal, 454 linux-sandbox.
//pkg/sql:sql_test                                                       FAILED in 25536.3s
  /home/michae2/.cache/bazel/_bazel_michae2/88dadb79cb9eca13f5ad7e068969d8c1/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/testlogs/pkg/sql/sql_test/test.log

Executed 1 out of 1 test: 1 fails locally.
INFO: Build Event Protocol files produced successfully.
INFO: Build completed, 1 test FAILED, 457 total actions
ERROR: exit status 3
```

sql.TestScrubFKConstraintFKMissing [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8091148?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8091148?buildTab=artifacts#/) on release-22.2 @ [ea0e966fe05a788fd123a08cec8ef5e935fa5496](https://github.com/cockroachdb/cockroach/commits/ea0e966fe05a788fd123a08cec8ef5e935fa5496):


```
=== RUN   TestScrubFKConstraintFKMissing
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestScrubFKConstraintFKMissing3112676340
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    scrub_test.go:605: unexpected error: pq: scrub-unique: inconsistent AS OF SYSTEM TIME timestamp; expected: 1671879728.062602212,1, got: 1671879728.062602212,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestScrubFKConstraintFKMissing3112676340
--- FAIL: TestScrubFKConstraintFKMissing (1.38s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScrubFKConstraintFKMissing.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/scrub_unique_constraint.go
+++ b/pkg/sql/scrub_unique_constraint.go
@@ -121,8 +121,7 @@ func (o *sqlUniqueConstraintCheckOperation) Start(params runParams) error {
 	}
 	asOf := ""
 	if o.asOf != hlc.MaxTimestamp {
-		asOf = fmt.Sprintf("AS OF SYSTEM TIME %[1]d",
-			o.asOf.WallTime)
+		asOf = fmt.Sprintf("AS OF SYSTEM TIME '%s'", o.asOf.AsOfSystemTime())
 	}
 	tableName := fmt.Sprintf("%s.%s", o.tableName.Catalog(), o.tableName.Table())
 	dup, _, err := duplicateRowQuery(o.tableDesc, o.cols, o.predicate, false /* limitResults */)

